src/client/client.py

```diff
@@ -17,197 +19,203 @@ class Client:
-            print(f"Check correct file name and path and try again ")
-
-
-    @staticmethod
-    def download(file_name: str):
-        """Download file from a server."""
-
-        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as client:
             try:
-                client.connect((Client.SERVER_IP, Client.SERVER_PORT))
+                reader, writer = await asyncio.open_connection(Client.SERVER_IP, Client.SERVER_PORT)
             except ConnectionRefusedError as e:
                 print(e)
                 print("Server probably ins't running!!!")
                 exit(1)
+ 
+            file_name = os.path.basename(file_path)
 
-            # Send DOWNLOAD request
-            client.send(f"DOWNLOAD;{file_name};{Client.get_username()}".encode())
+            # Send UPLOAD request
+            writer.write(f"UPLOAD;{file_name};{Client.get_username()}{Client.EOM}".encode())
 
-            # Receive aswer: 'OK;Authenticated' or 'ERROR;FileNotFoundError' or 'ERROR;NotAuthenticatedError'
-            answer = client.recv(Client.BUFFER_SIZE).decode()
+            # Receive aswer
+            data = await reader.readuntil(Client.EOM.encode())
+            answer = data.decode()
 
             if 'OK' in answer:
                 # Answer: 'OK;Authenticated'
-                file_info = client.recv(Client.BUFFER_SIZE).decode()
-                file_name, file_size = file_info.split(';')
+                file_size = os.path.getsize(file_path)
+                progress = tqdm.tqdm(range(file_size), f"Sending {file_name}", unit="B", unit_scale=True, unit_divisor=1024)
 
-                progress = tqdm.tqdm(range(int(file_size)), f"Receiving {file_name}", unit="B", unit_scale=True, unit_divisor=1024)
-
-                # Receive file
-                with open(str(os.path.join(os.path.expanduser('~/Downloads'), file_name)), "wb") as f:
+                # Send file
+                with open(file_path, "rb") as f:
                     while True:
-                        bytes_read = client.recv(Client.BUFFER_SIZE)
+                        bytes_read = f.read(Client.BUFFER_SIZE)
                         if not bytes_read:
                             break
-                        f.write(bytes_read)
                         progress.update(len(bytes_read))
+                        writer.write(bytes_read)
+                        await writer.drain()
 
-            elif 'FileNotFoundError' in answer:
-                # Answer: 'ERROR;FileNotFoundError'
-                print(f"File '{file_name}' does NOT exist, CANNOT download")
-                print("Available files:")
-                Client.list_files(detailed=False)
-
-            elif 'NotAuthenticatedError' in answer:
-                # Answer: 'ERROR;NotAuthenticatedError'
-                print(f"User '{Client.get_username()}' can't be authenticated, CANNOT downdload")
-
+            elif 'NotAuthenticated' in answer:
```

Comment:
Tato možnost nastane v případě, kdy se o nahrání souboru snaží uživatel s jménem co není v databázi?

Reply:
Ano ano, přesně tak. je to zasláno klientovi který poslal v reqestu jméno které neexistuje v databázi